### PR TITLE
Add access origin by weight rate lambda edge app

### PR DESCRIPTION
**Pull request by dengmingtong**
*Issue #, if available:*
No

*Description of changes:*
Add access origin by weight rate lambda edge app

*How Has This Been Tested:*
sam package and sam publish in my account

*[x] My testing has passed*
passed

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: edge/nodejs/access-origin-by-weight-rate/README.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+The SAM CLI installs dependencies defined in `access-origin-by-weight-rate/package.json`, creates a deployment package, and saves it in the `.aws-sam/build` folder.
+
+Test a single function by invoking it directly with a test event. An event is a JSON document that represents the input that the function receives from the event source. Test events are included in the `events` folder in this project.
+
+Run functions locally and invoke them with the `sam local invoke` command.
+
+```bash
+access-origin-by-weight-rate$ sam local invoke SimpleLambdaEdgeFunction --event events/event.json
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/README.md:62]: Better to update SimpleLambdaEdgeFunction to your own function

File: edge/nodejs/access-origin-by-weight-rate/README.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+The application template uses AWS Serverless Application Model (AWS SAM) to define application resources. AWS SAM is an extension of AWS CloudFormation with a simpler syntax for configuring common serverless application resources such as functions, triggers, and APIs. For resources not included in [the SAM specification](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md), you can use standard [AWS CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-template-resource-type-ref.html) resource types.
+
+## Fetch, tail, and filter Lambda function logs
+
+To simplify troubleshooting, SAM CLI has a command called `sam logs`. `sam logs` lets you fetch logs generated by your deployed Lambda function from the command line. In addition to printing the logs on the terminal, this command has several nifty features to help you quickly find the bug.
+
+`NOTE`: This command works for all AWS Lambda functions; not just the ones you deploy using SAM.
+
+```bash
+access-origin-by-weight-rate$ sam logs -n HelloWorldFunction --stack-name access-origin-by-weight-rate --tail
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/README.md:76]: Update HelloWorldFunction to your function name

File: edge/nodejs/access-origin-by-weight-rate/access-origin-by-weight-rate/package.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "simple-lambda-edge",
+  "version": "1.0.0",
+  "description": "Simple lambda@edge sample for NodeJS",
+  "main": "app.js",
+  "repository": "https://github.com/cc4i/aws-cloudfront-extensions/edge/nodejs/simple-lambda-edge",
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/access-origin-by-weight-rate/package.json:6]: Replace simple-lambda-edge to access-origin-by-weight-rate

File: edge/nodejs/access-origin-by-weight-rate/access-origin-by-weight-rate/package.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "simple-lambda-edge",
+  "version": "1.0.0",
+  "description": "Simple lambda@edge sample for NodeJS",
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/access-origin-by-weight-rate/package.json:4]: Update description

File: edge/nodejs/access-origin-by-weight-rate/access-origin-by-weight-rate/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "simple-lambda-edge",
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/access-origin-by-weight-rate/package.json:2]: Update name to access-origin-by-weight-rate

File: edge/nodejs/access-origin-by-weight-rate/template.yaml [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  Function:
+    Timeout: 30
+    Tags:
+      Publisher: GCR-Solutions
+
+Metadata:
+  AWS::ServerlessRepo::Application:
+    Name: access-origin-by-weight-rate
+    Description: SAM Template for access-origin-by-weight-rate
+    Author: GCR Solutions
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/template.yaml:20]: Update to AWS

File: edge/nodejs/access-origin-by-weight-rate/template.yaml [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+                - edgelambda.amazonaws.com
+            Action: sts:AssumeRole
+        ManagedPolicyArns:
+          - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+          - arn:aws:iam::aws:policy/AWSXrayWriteOnlyAccess
+          - arn:aws:iam::aws:policy/AWSLambdaFullAccess
```
Comment (NingLu) [inline edge/nodejs/access-origin-by-weight-rate/template.yaml:61]: Is AWSLambdaFullAccess really needed here?

File: edge/nodejs/access-origin-by-weight-rate/template.yaml (hunk 6 above)
Comment (dengmingtong) [reply edge/nodejs/access-origin-by-weight-rate/template.yaml:20]: Keep on GCR Solutions until getting the permission to use "AWS"

**Review by NingLu — APPROVED** [review]

**Pull request merged**